workflow/tools/fastq_to_sam.py

```diff
@@ -88,6 +88,12 @@
         except Exception as e:
             sys.exit('\n%s: FastqToSam failed: %s and %s' % (e, file_with_path[0], file_with_path[1]))
 
+        for f in file_with_path:
+            try:
+                os.remove(f)
+            except Exception as e:
+                sys.exit('\n%s: Delete file failed: %s' % (e, f))
+
```

Comment:
Don't delete input file within this loop `for rg in readGroups:`, instead, wait until all RGs are done.

Same comment for those steps with deletion.